### PR TITLE
Avoid array copy in generating process builder commands 

### PR DESCRIPTION
**Pull request by bowenliang123**
# :mag: Description
## Issue References 🔗

As described.

## Describe Your Solution 🔧
 
Currently, each process builder use `ArrayBuffer.toArray` for generating process builder commands which includes unnecessary array copy. Reuse the buffer and use `Iteratable` for commands.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior Without This Pull Request :coffin:
No behavior change.

#### Behavior With This Pull Request :tada:
No behavior change.

#### Related Unit Tests
No behavior change.

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala [hunk 1]
```diff
@@ -142,7 +142,7 @@ trait ProcBuilder {
   }
 
   final lazy val processBuilder: ProcessBuilder = {
-    val pb = new ProcessBuilder(commands: _*)
+    val pb = new ProcessBuilder(commands.toStream.asJava)
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:145]: will this introduce the overhead?

**Comment by pan3793** [comment]
A generally principle: for performance in-sensitive code path, readability is the most important thing

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala (hunk 1 above)
Comment (bowenliang123) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:145]: `toStream` uses a iterator, which prevents array copy to a list.

Comment (pan3793) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:145]: it requires to be converted to Array eventually, so early transformation should be good too. would prefer to keep as-is

Comment (bowenliang123) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:145]: OK, I got your point. Closing this PR.

**Comment by bowenliang123** [comment]
Reopened this PR after offline discussions.

Previously: ArrayBuffer - copyTo Array-> _* -> copyTo ArrayList (when constructing ProcessBuilder)
With PR: ArrayBuffer - asStream-> toIterator -> construct with List  (when constructing ProcessBuilder)

This PR helps to prevent first stage `ArrayBuffer - copyTo Array` array copying.

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**